### DISCUSSION TITLE
Add AssemblyLoadContext to .NET Standard

### DISCUSSION THREAD

**Discussion by techfan101** [dotnet/runtime#46815] [Ideas] [answered]
I'm intrigued by the `AssemblyLoadContext` class.  It provides a cleaner mechanism for isolating assemblies than those provided by `AppDomain`.  I'd love to take advantage of it in a VSIX package I'm developing, but I'm prevented from doing so because it's limited to .NET Core.   Please consider including this class in .NET Standard.

Are there any plans to include it in some future version of .NET Standard?  Is there any (approximate) timeline for this inclusion?

**Comment by PathogenDavid** [comment, marked as answer]
Short version:

> Are there any plans to include it in some future version of .NET Standard?

No

> Is there any (approximate) timeline for this inclusion?

Never

-------------------

.NET Standard is done. .NET Standard 2.0 was the last version that supported .NET Framework, [.NET Standard 2.1](https://devblogs.microsoft.com/dotnet/announcing-net-standard-2-1/) is the last version that will be released in general. (See here: [The future of .NET Standard](https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/).)

Furthermore, in order to get `AssemblyLoadContext` added to .NET Standard in a way which a Visual Studio extension could use it, you would need `AssemblyLoadContext` to be added to .NET Framework. [.NET Framework 4.8 is the last version of .NET Framework](https://devblogs.microsoft.com/dotnet/net-core-is-the-future-of-net/). It is only receiving fixes, no new features.

**Comment by techfan101** [comment]
@PathogenDavid Thank you. That does seem final.  I guess I'll back-port pieces of `AssemblyDependencyResolver` to work with `AppDomain`.  Much as I hate duplicating the effort, I don't expect VSIX packages (or the associated project templates) are going to catch up with .NET 5.0 anytime in the near future :(

**Reply by PathogenDavid** [reply]
No problem! And yeah, it's definitely not ideal. I would also not hold my breath on Visual Studio switching to .NET 5.0 either. Although it's holding back Roslyn analyzers and source generators too, so I like to think we'll see it some day eventually.

I don't know your usage scenario, but if it's possible to run your code in a .NET 5 process outside of Visual Studio and communicate with it via IPC, you might consider doing that instead.

**Reply by techfan101** [reply]
@PathogenDavid I previously considered a separate process to get better isolation than `AppDomain` provides.  Back then, I ruled it out as unnecessarily complicated.  Though, as you suggest, perhaps it's time to revisit that design decision.  In fact, it may be time to reconsider the project in its entirety.  It was supposed to be a fun midnight hack.  It might be time to return to less "fun" pursuits :)

**Reply by PathogenDavid** [reply]
Ah, see there's the problem. You tried to combine fun and Visual Studio extension development 😅
